src/testdir/test_termdebug.vim

```diff
@@ -300,4 +300,41 @@ func Test_termdebug_mapping()
   %bw!
 endfunc
 
+func Test_termdebug_bufnames()
+  " Test if user has filename/folders named gdb, Termdebug-gdb-console,
+  " etc. in the current directory
+  let g:termdebug_config = {}
+  let g:termdebug_config['use_prompt'] = 1
+  let filename = 'gdb'
+  let replacement_filename = 'Termdebug-gdb-console'
+
+  call writefile(['This', 'is', 'a', 'test'], filename, 'D')
+  " Throw away the file once the test has done.
+  Termdebug
+  call WaitForAssert({-> assert_equal(3, winnr('$'))}
+  " A file named filename already exists in the working directory,
+  " hence you must call the newly created buffer differently
+  call WaitForAssert({-> assert_false(bufexists(filename))})
+  call WaitForAssert({-> assert_true(bufexists(replacement_filename))})
+  quit!
+  call WaitForAssert({-> assert_equal(1, winnr('$'))})
+
+  " Check if error message is in :message
+  let g:termdebug_config['disasm_window'] = 1
+  let filename = 'Termdebug-asm-listing'
+  call writefile(['This', 'is', 'a', 'test'], filename, 'D')
+  " Check only the head of the error message
+  let error_message = "You have a file/folder named '" .. filename .. "'"
+  Termdebug
+  sleep 2
```

Comment:
Can you replace this hard-coded sleep with WaitForAssert()?

Reply:
Done! Sorry for so many commits, it's my first time. I wish I could test the CI on my fork first... shall I cancel the PR if the CI still fails?

Reply:
No worries. After all, that's what we have CI for ;) BTW: you can also run the tests locally: https://github.com/vim/vim/blob/master/src/testdir/README.txt

Just need to run `make test_termdebug.res` and then check the `test.log` and `messages` file.

Reply:
AWESOME! I am going to give it a shot straight away. :)

Reply:
From the README

```
RUNNING THE TESTS:

To run a single test from the src directory:

    $ make test_<name>

The below commands should be run from the src/testdir directory.

To run a single test:

    $ make test_<name>.res

```

I am in `./vim/src/testdir` but when I run make `test_termdebug.res` I get `make: *** No rule to make target '../vim', needed by 'test_termdebug.res'.  Stop.` Perhaps there is some requirement that I am missing?

Reply:
go up and directory and make vim first.

Reply:
Thanks! 
I built the `vim` executable. The test run but it seems that my tests are not run:

```
$ make test_termdebug.res
VIMRUNTIME=../../runtime  ../vim -f  -u unix.vim --gui-dialog-file guidialog -U NONE --noplugin --not-a-term -S runtest.vim test_termdebug.vim --cmd 'au SwapExists * let v:swapchoice = "e"' | LC_ALL=C LANG=C LANGUAGE=C awk '/Executing Test_/{match($0, "([0-9][0-9]:[0-9][0-9] *)?Executing Test_[^\\)]*\\)"); print substr($0, RSTART, RLENGTH) "\r"; fflush()}'
02:05 Executing Test_termdebug_basic()
02:09 Executing Test_termdebug_bufnames()
02:09 Executing Test_termdebug_mapping()
02:09 Executing Test_termdebug_tbreak() 
```

It is missing `Test_termdebug_bufnames()`. Shall I add it to some other file?